granulate_utils/linux/cgroups/memory_controller.py

```diff
@@ -0,0 +1,105 @@
+#
+# Copyright (c) Granulate. All rights reserved.
+# Licensed under the AGPL3 License. See LICENSE.md in the project root for license information.
+#
+
+from abc import abstractmethod
+from pathlib import Path
+from typing import Optional, Union
+
+from granulate_utils.linux.cgroups.base_controller import BaseController
+from granulate_utils.linux.cgroups.cgroup import CgroupCore, CgroupCoreV1, CgroupCoreV2
+
+
+class MemoryControllerInterface:
+    memory_limit: str
+    usage: str
+    max_usage_in_bytes: str = ""
+    swap_limit: str = ""
+
+    def __init__(self, cgroup: CgroupCore):
+        self.cgroup = cgroup
+
+    @abstractmethod
+    def set_limit_in_bytes(self, limit: int) -> None:
+        pass
+
+    def get_memory_limit(self) -> int:
+        return int(self.cgroup.read_from_interface_file(self.memory_limit))
+
+    def _set_swap_limit(self, limit: str) -> None:
```

Comment:
`limit: int`. I know that with `limit: str` you're able to reuse the same function for v1 & v2, but sry, it's a hack, the logic is different